### PR TITLE
Merge QoL changes addressing #21

### LINKED ISSUE #19: Toggle visibility in Filter List
Add an option to toggle individual filters on and off, much like Photoshop layers. Useful for debugging and reducing clutter without having to redo settings. Also optionally move delete filter button to each list item to reduce the number of clicks per operation.

### LINKED ISSUE #21: Instructions on default page
When running an insitu node (rosrun insitu insitu), it pops up with a blank page.
As a novice, I clicked around and crashed it.

Suggest instead of a blank page, we make it load a text file by default. This text file can have instructions/overview of how to use add modes -> views -> filter.

@PaperFanz @AdamPettinger

### LINKED ISSUE #23: lockToImageSize() bug
**Bug description**
Setting `lockToImageSize ()` to return "true" doesn't return the filter width() and height() in the `apply()` function

**To Reproduce**
1. Go to any filter.hpp file
2. Under public field in the class definition type:

`   bool lockToImageSize(void)
    {
        return true;
    }`
    
3.  Use width() or height() in the  `apply()` function. This returns 0. 
eg. `std::cout << width() <<std::endl; `

**Expected behavior**
`std::cout << width() <<std::endl; ` should return the correct root image width and height. 

 

**Desktop (please complete the following information):**
 - Ubuntu 20.04 
 - ROS Version: Noetic

### PR DESCRIPTION
**Pull request by PaperFanz**
Changes:
- tutorial page on blank task view creation
- fix crashes from clicking random menu items
- start `insitu_utils` subpackage for common filter functions

### CODE REVIEW

File: insitu/src/main_window.cpp [hunk 1]
```diff
@@ -164,6 +166,14 @@ void MainWindow::restore(Json::Value& json)
...
     }
     tabmanager->setCurrentIndex(json.get("currentMode", 0).asInt());
+
+    /* show tutorial page if no modes are restored */
+    if (ui.tabmanager->count() == 0)
+    {
+        QString name = "Tutorial";
+        ModeContainer* tutorialMode = new ModeContainer(name);
+        ui.tabmanager->addTab(tutorialMode, name);
+    }
```
Comment (AdamPettinger) [inline insitu/src/main_window.cpp:182]: Excellent

File: insitu_utils/package.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  <name>insitu_utils</name>
+  <version>0.1.0</version>
+  <description>
+
+    insitu_utils
+
+  </description>
+  <maintainer email="sci@gmail.com">sci</maintainer>
+  <author>sci</author>
+  <license>BSD</license>
```
Comment (AdamPettinger) [inline insitu_utils/package.xml]: This is public, we should fill these out

**Review by AdamPettinger — COMMENTED** [review]
Looks good, with one minor detail. Will test quickly and approve

File: insitu_utils/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+catkin_package(
+    LIBRARIES ${PROJECT_NAME}
+    INCLUDE_DIRS inc lib
+    CATKIN_DEPENDS insitu
+    DEPENDS OpenCV
+)
+
+include_directories(
+    inc
```
Comment (AdamPettinger) [inline insitu_utils/CMakeLists.txt]:
```suggestion
```

File: insitu_utils/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+# CATKIN
+find_package(catkin REQUIRED)
+find_package(OpenCV 4.2 REQUIRED)
+
+include_directories(${catkin_INCLUDE_DIRS})
+
+catkin_package(
+    LIBRARIES ${PROJECT_NAME}
+    INCLUDE_DIRS inc lib
```
Comment (AdamPettinger) [inline insitu_utils/CMakeLists.txt]:
```suggestion
    INCLUDE_DIRS lib
```

**Review by AdamPettinger — CHANGES_REQUESTED** [review]
Came across an issue building, probably from the recent merge. See suggestions.

Other few things:

- The top level CMakeLists links to `/opt/melodic/...`, not noetic
- When I pulled and built this, then did `roslaunch usb_cam usb_cam` and `rosrun insitu insitu`, I actually didn't get the tutorial page (just an empty insitu mode). Maybe that is because it was reading an old config file automatically. Eventually I did get the tutorial page by saving a new file (when the modes were empty) and when I reloaded it, the tutorials came up. Thoughts on this?

**Comment by PaperFanz** [comment]
> 
> 
> Came across an issue building, probably from the recent merge. See suggestions.
> 
> Other few things:
> 
>     * The top level CMakeLists links to `/opt/melodic/...`, not noetic
> 
>     * When I pulled and built this, then did `roslaunch usb_cam usb_cam` and `rosrun insitu insitu`, I actually didn't get the tutorial page (just an empty insitu mode). Maybe that is because it was reading an old config file automatically. Eventually I did get the tutorial page by saving a new file (when the modes were empty) and when I reloaded it, the tutorials came up. Thoughts on this?

For the CMakeLists.txt, that will be fixed in a commit.

For the tutorial, that might be the case. It should be a simple fix. Thanks for catching that!

**Review by PaperFanz — COMMENTED** [review]
Should have addressed all the issues with the last two commits, @AdamPettinger could you re-review?

**Review by AdamPettinger — APPROVED** [review]
Looks great!

**Pull request merged**